ethereum/rpc/backend/backend.go

```diff
@@ -244,14 +244,14 @@ func (e *EVMBackend) HeaderByNumber(blockNum types.BlockNumber) (*ethtypes.Heade
 
 	req := &evmtypes.QueryBlockBloomRequest{}
 
-	res, err := e.queryClient.BlockBloom(types.ContextWithHeight(resBlock.Block.Height), req)
+	blockBloomResp, err := e.queryClient.BlockBloom(types.ContextWithHeight(resBlock.Block.Height), req)
 	if err != nil {
 		e.logger.Warningf("HeaderByNumber BlockBloom fail %d", resBlock.Block.Height)
-		return nil, err
+		blockBloomResp = &evmtypes.QueryBlockBloomResponse{Bloom: ethtypes.Bloom{}.Bytes()}
```

Comment:
This is consistent with `getBlockByNumber` which never throw error when blockBloom cannot be found but return empty bloom.